const promiseOne = new Promise(function(resolve,reject){
    //do an async task
    //DB Calls,cryptography,network 

    setTimeout(function(){
        console.log('Async task is complete');
        resolve()
    },1000)
    })

    promiseOne.then(function(){
        console.log("Promise consumed");
    })

    new Promise(function(resolve,reject){
        setTimeout(function(){
            console.log("Async task 2");
            resolve()
        },1000)
    }).then(function(){
        console.log("Async 2 resolved");
        
    })

    const PromiseThree= new Promise(function(resolve,reject){
            setTimeout(function(){
                resolve({username: "Chai",email: "chai@example.com"})
            },1000)
    })

    PromiseThree.then(function(user){
        console.log(user);
    })

    const PromiseFour=new Promise(function(resolve,reject){
        setTimeout(function(){
            let error = false
            if(!error){
                resolve({username:"hitesh",password:"123"})
            }else{
                reject('ERROR: Something went wrong')
            }
        
        },1000)
    })
  PromiseFour
  .then((user)=>{
        console.log(user);
        return user.username
    })
    .then((username)=>{
        console.log(username);
    })
    .catch(function(error){
        console.log(error);  
    })
    .finally(() => console.log("The promise is either rsolved or rejected"));
    
const promiseFive= new Promise(function(resolve,reject){
    setTimeout(function(){
            let error = true
            if(!error){
                resolve({username:"Javascript",password:"123"})
            }else{
                reject('ERROR: JS went wrong')
            }
        
        },1000)
});
async function consumePromiseFive(){
   try{ await promiseFive
    console.log(Response);
   }catch(error){
    console.log(error);
    
   }
}
consumePromiseFive()

/* async function getAllUsers(){
    try{
        fetch('https://jsonPlaceholder.typiode.com/users')
    const data= Response.json()
    console.log(Response);
    console.log(data);
    }catch(error){
        console.log("e: eRROR");
        
    }
    
} */
//getAllUsers()

fetch('https://jsonPlaceholder.typiode.com/users')
.then((response)=> {
    return response.json()
})
.then((data)=> {
    console.log(data);
    
})
.catch((error)=> console.log(error))
